extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -500,19 +506,42 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
     }
   }
 
+  /** @override */
+  delayAdRequestEnabled() {
+    if (this.element.getAttribute(LAZY_FETCH_ATTRIBUTE) !== 'true') {
+      return false;
+    }
+    return getAmpAdRenderOutsideViewport(this.element) || 3;
```

Comment:
This line is the same as the equivalent code in `amp-ad-network-adsense-impl.js`.  What would you think of refactoring the amp-a4a API to:

```
delayAdRequestEnabled() -> true/false
delayedAdRequestMayContinue() -> weird true / false / number thing
```

Then amp-a4a could contain:

```
delayAdRequestEnabled() {
  return false;
}
delayedAdRequestMayContinue() {
  return getAmpAdRenderOutsideViewport(this.element) || 3;
}
```

And adsense could override `delayAdRequestEnabled` to be unconditionally true while doubleclick could override it to check for `data-lazy-fetch`?